src/components/Charts/ChartRt.tsx

```diff
@@ -0,0 +1,265 @@
+const hasData = (d: any) =>
+  !isUndefined(x(d)) &&
+  !isUndefined(yRt(d)) &&
+  !isUndefined(yLow(d)) &&
+  !isUndefined(yHigh(d));
+
+const ChartRt = ({
+  projectionDataset,
+  width,
+  height = 400,
+  marginTop = 5,
+  marginBottom = 40,
+  marginLeft = 40,
+  marginRight = 5,
+}: {
+  projectionDataset: ProjectionDataset;
+  width: number;
+  height?: number;
+  marginTop?: number;
+  marginBottom?: number;
+  marginLeft?: number;
+  marginRight?: number;
+}) => {
+  const chartWidth = width - marginLeft - marginRight;
+  const chartHeight = height - marginTop - marginBottom;
+
+  const data = projectionDataset.data.filter(hasData);
+
+  // TODO(@pnavarrc): This is so TS doesn't complain
+  const minDate = d3min(data, x) || new Date('2020-01-01');
+  const maxDate = CHART_END_DATE;
+
+  const xScale = scaleTime({
+    domain: [minDate, maxDate],
+    range: [0, chartWidth],
+  });
+
+  const yDataMin = 0;
+  const yDataMax = d3max(data, yRt);
+
+  const yScale = scaleLinear({
+    domain: [yDataMin, yDataMax],
+    range: [chartHeight, 0],
+  });
+
+  const xCoord = (d: any) => xScale(x(d));
+  const yCoord = (d: any) => yScale(yRt(d));
+
+  const yTicks = computeTickPositions(yDataMin, yDataMax, CASE_GROWTH_RATE);
+  const yAxisTicks = _tail(yTicks);
+
+  const { x: lastValidDate } = last(data);
+  const dateTruncation = getTruncationDate(lastValidDate, RT_TRUNCATION_DAYS);
+  const prevData = data.filter((d: any) => x(d) <= dateTruncation);
+  const restData = data.filter((d: any) => x(d) >= dateTruncation);
+  const truncationDataPoint = last(prevData);
+
+  const regions = getChartRegions(yDataMin, yDataMax, CASE_GROWTH_RATE);
+  const truncationZone = getZoneByValue(
+    yRt(truncationDataPoint),
+    CASE_GROWTH_RATE,
+  );
+
+  const { tooltipData, tooltipOpen, showTooltip, hideTooltip } = useTooltip();
+  const onMouseOver = (
+    event: React.MouseEvent<SVGPathElement, MouseEvent>,
+    d: any,
+  ) => {
+    // @ts-ignore - typing bug
+    const coords = localPoint(event.target.ownerSVGElement, event);
+    showTooltip({
+      tooltipLeft: coords?.x || 0,
+      tooltipTop: coords?.y || 0,
+      tooltipData: d,
+    });
+  };
+
+  const mainClipPathId = randomizeId('chart-clip-path');
+  return (
+    <Style.ChartContainer>
+      <svg width={width} height={height}>
+        <Group left={marginLeft} top={marginTop}>
+          <RectClipPath
+            id={mainClipPathId}
+            width={chartWidth}
+            height={chartHeight}
+          />
+          <Group clipPath={`url(#${mainClipPathId})`}>
+            <Style.SeriesArea>
+              <Area
+                data={data}
+                x={xCoord}
+                y0={(d: any) => yScale(yLow(d))}
+                y1={(d: any) => yScale(yHigh(d))}
+                curve={curveNatural}
+              />
+            </Style.SeriesArea>
+            {regions.map((region, i) => {
+              const clipPathZoneId = randomizeId(`clip-region-${region.name}`);
```

Comment:
I think a comment here about what this is doing would be helpful

Reply:
I created a component to abstract the clipping behavior and added an explanation of how it works, I hope it's more clear now.